### PR TITLE
Add an extra note about np.nan etc

### PR DESCRIPTION
**Pull request by hyanwong**

### CODE REVIEW

File: python/tskit/util.py [hunk 1]
```diff
@@ -60,7 +60,10 @@ def is_unknown_time(time):
     """
     As the default unknown mutation time (:const:`UNKNOWN_TIME`) is a specific NAN value,
     equality always fails. This method compares the bitfield such that unknown times can
-    be detected. Either single floats can be passed or lists/arrays.
+    be detected. Either single floats can be passed or lists/arrays. Note that many types
```
Comment (jeromekelleher) [inline python/tskit/util.py:63]: It's a bit more accurate to say that there are many different NaN *values*, but only one specific NaN matches UNKNOWN_TIME. Maybe link to the NaN wikipedia page so people can read about it: https://en.wikipedia.org/wiki/NaN

**Review by jeromekelleher — CHANGES_REQUESTED** [review]
Slightly misleading at the moment IMO

**Review by benjeffery — APPROVED** [review]
I've fixed this up. Hopefully makes sense!

**Comment by hyanwong** [comment]
Great, thanks @benjeffery

**Review by jeromekelleher — APPROVED** [review]

**Comment by benjeffery** [comment]
@mergifyio rebase

**Pull request merged**